tests/conftest.py

```diff
@@ -17,9 +17,7 @@
 ]
 
 SESSION = sessionmaker()
-TEST_DATABASE_URL = os.environ.get(
-    "TEST_DATABASE_URL", "postgresql://postgres@localhost:5433/lms_test"
-)
+TEST_DATABASE_URL = os.environ["TEST_DATABASE_URL"]
```

Comment:
I get why this is done, but this is going to make it a collosal pain for me to run any tests as they no longer work without specifying the env var.

This is fine if you run the tests through "make test" at all times, but if you use an IDE like me, or you run the tests using pytest on the command line this is very unhelpful.

Having a different defaulting `TEST_DATABASE_URL` in the `functional/conftest.py`, `unit/conftest.py` and `bdd/environment.py` is easy enough, but we need to stop directly reading this variable into `TEST_SETTINGS`.

A pytest fixture won't work here as the BDD stuff isn't pytest based. I think we can do this by changing this to `get_test_settings(db_url)` or something similar. It's only used in three places from what I can see, so that shouldn't be too hard?

I've also spotted that two of the three places is adding "session_cookie_secret" to the settings, so maybe we should chuck them in the main config and be done with it.